### PR TITLE
Print executable name on cargo test --no-run #2

### LINKED ISSUE #9957: `cargo test --no-run`  should print the resulting binary name
Problem:

I am profiling my Rust application, so I want to compile the tests with `--no-run` and run the binary directly. However, the output doesn't say where the result is:

```console
$ cargo test --no-run --release --package runtime-tester --features nearcore/no_cache --lib -- perf_cliff --exact --nocapture
   Compiling runtime-tester v0.1.0 (/home/matklad/projects/nearcore/test-utils/runtime-tester)
    Finished release [optimized] target(s) in 5.49s
$
```

I have to manually `ls traget/release/deps` to get the file name

Proposed solution:

The name is convenientl printed when you don't pass no-run:

```console
$ cargo test --release --package runtime-tester --features nearcore/no_cache --lib -- perf_cliff --exact --nocapture
    Finished release [optimized] target(s) in 0.13s
     Running unittests (target/release/deps/runtime_tester-160c2544f9ae8411)
```

Would be great to add a message similar to `Running target (path)` to `--no-run` output.

### PR DESCRIPTION
**Pull request by yerke**
Closes #9957

This is a continuation of https://github.com/rust-lang/cargo/pull/9959.

### CODE REVIEW

**Comment by ehuss** [comment]
This is marked as a draft, is there some additional work you want to do?

**Comment by yerke** [comment]
@ehuss Yes, you mentioned that you wanted benches to be displayed differently in the other PR. I want to get to that shortly.
I have another question for you. Do you want this functionality (naming executables) to work in verbose mode too? I don’t think it does right now.

File: src/cargo/ops/cargo_test.rs [hunk 1]
```diff
@@ -264,3 +255,77 @@ fn run_doc_tests(
...
+
+fn cmd_builds(
+    config: &Config,
+    cwd: &Path,
+    unit: &Unit,
+    path: &PathBuf,
+    script_meta: &Option<Metadata>,
+    test_args: &[&str],
+    compilation: &Compilation<'_>,
+    exec_type: &str,
```
Comment (yerke) [inline src/cargo/ops/cargo_test.rs:302]: @ehuss Let me know if you want this to be an enum.

**Comment by ehuss** [comment]
Thanks, looks great!

FCP finished in https://github.com/rust-lang/cargo/pull/9959#issuecomment-952236631.

@bors r+

**Pull request merged**